storage/mysql/log_storage.go

```diff
@@ -292,9 +293,9 @@ func (t *logTreeTX) DequeueLeaves(ctx context.Context, limit int, cutoffTime tim
 	for rows.Next() {
 		var leafIDHash []byte
 		var merkleHash []byte
-		var meta dequeueMeta
+		var queueTimestamp int64
```

Comment:
QueueTimestamp is a new column, so I suppose it must be nullable (because backwards compatibility). In this case you should be using sql.NullInt64 as the type.

Incidentally, could you make sure we have tests that dequeue leaves with null timestamps?

Reply:
Actually no, you can create new `NOT NULL` columns, and those cells will take the default value of the type.

Reply:
Are you sure about the not null? I recall problems with that, but maybe I have a different DB in mind.

Either way, it may be better to distinguish null from zero, though, so instead of returning QueueTime/IntegrateTime = epoch for "old" leaves we can return nothing.

Reply:
Seems to be the case (e.g. [this](https://stackoverflow.com/questions/22868345/mysql-add-a-not-null-column))

We can reasonably safely special case when the timestamps are zero (that's impossible given when we started writing Trillian, so we know that it's just old data), but gain the advantage of it being impossible to create more leaves with no timestamps.